File: flink-monitor/src/test/java/com/lfmunoz/monitor/actions/consumer/KafkaConsumerUtils.kt
package com.lfmunoz.monitor.actions.consumer

import com.lfmunoz.flink.web.WsPacket
import com.lfmunoz.flink.web.WsPacketCode
import com.lfmunoz.flink.web.WsPacketType
import com.lfmunoz.monitor.actions.producer.KafkaProducerConfig
import com.lfmunoz.monitor.actions.producer.KafkaProducerDTO
import com.lfmunoz.monitor.actions.producer.KafkaProducerType
import com.lfmunoz.monitor.actions.producer.buildWsPacket
import kotlin.random.Random

fun buildKafkaSamplePkt(debounce: Long): WsPacket {
  val dto = KafkaConsumerDTO(type = KafkaConsumerType.SAMPLE, body = debounce.toString())
  return createWsPacketForKafka(dto)
}

fun buildConfigWritePkt(config: KafkaConsumerConfig): WsPacket {
  val dto = KafkaConsumerDTO( type = KafkaConsumerType.CONFIG_WRITE, body = config.toJson())
  return createWsPacketForKafka(dto)
}

fun buildConfigReadPkt(): WsPacket {
  val dto = KafkaConsumerDTO( type = KafkaConsumerType.CONFIG_READ, body = "")
  return createWsPacketForKafka(dto)
}

fun createWsPacketForKafka(dto: KafkaConsumerDTO): WsPacket {
  return WsPacket(
    payload = dto.toJson(),
    action = WsPacketType.KAFKA_CONSUMER,
    id = Random.nextInt(),
    code = WsPacketCode.REQ
  )
}

File: webapp/src/actions/kafkaConsumer/KafkaConsumerUtils.js
/**
 *  TestUtils.js
 */

import WsPacket from "@/websocket/WsPacket.js";
import { generateRandomInt32, Action, Code} from "@/websocket/ClientUtils.js"

// ________________________________________________________________________________
// KAFKA ACTION
// ________________________________________________________________________________
export function createWsPacketForKafka(obj) {
    const aWsPacket = new WsPacket({
        payload: JSON.stringify(obj),
        action: Action.KAFKA_CONSUMER,
        id: generateRandomInt32(),
        code: Code.REQ
    })
    return aWsPacket
}

export const KafkaProducerType = {
    SAMPLE: "SAMPLE",
    CONFIG_READ: "CONFIG_READ",
    CONFIG_WRITE: "CONFIG_WRITE"
}

export function buildKafkaReadConfig() {
    const dto = { 
        'type': KafkaProducerType.CONFIG_READ, 
        'body': '' 
    }
    return createWsPacketForKafka(dto)
}

export function buildKafkaWriteConfig(body) {
    const dto = { 
        'type': KafkaProducerType.CONFIG_WRITE, 
        'body': JSON.stringify(body)
    }
    return createWsPacketForKafka(dto)
}

export function buildKafkaSample(debounce) {
    const dto = { 
        'type': KafkaProducerType.SAMPLE, 
        'body': debounce
    }
    return createWsPacketForKafka(dto)
}